### DISCUSSION TITLE
Intel SGX enclave support documentations

### DISCUSSION THREAD

**Discussion by AnandakumarNatarajan** [codenotary/immudb#1190] [Q&A]
Hello Team,

https://codenotary.com/technologies/immudb/

![image](https://user-images.githubusercontent.com/16556974/164382240-e0aae5ae-6a48-4d1b-b9ba-535ddd462ce7.png)

On the Immudb home page, I could see references about the support for Intel SGX enclaves. But, in the documentation, I couldn't anything related to it. 

Share the documents that are related to Intel SGX and Immudb if any are available.

Thanks
Anand

**Comment by jeroiraz** [comment]
Thanks for reaching out @AnandakumarNatarajan. So far we had only made custom integrations with SGX, keeping the current state securely stored in the enclave. Could you describe your use-case?

**Reply by AnandakumarNatarajan** [reply]
No specific use case. I just wanted to know how & what you guys doing with Intel SGX.

Maybe share the details about the custom integrations for understanding.

Thanks
Anand

**Reply by AnandakumarNatarajan** [reply]
Also, I would like to understand how transaction indexing is done in the Immudb. 

will be helpful if you share document links to understand Immudb internals.

**Reply by jeroiraz** [reply]
thanks.

Regarding SGX, you may find some initial work under here `https://github.com/codenotary/immudb/tree/intel_sgx`, where the server state was securely stored and updated only when proofs are validated.

Sure, we are going to document how indexing works and its compaction procedure very soon

**Comment by AnandakumarNatarajan** [comment]
![image](https://user-images.githubusercontent.com/16556974/170421609-09aed2d6-ec20-4944-a65f-d16fedc68bb8.png)

I was casually checking whether any updates are added regarding the Intel SGX and Indexing. I found Intel SGX keywords are removed from the home page. Is the "Everything is encrypted" refers to Intel SGX?
